### PR TITLE
Add example examples

### LINKED ISSUE #96492: Reduce ceremony with small samples
Some of our examples have quite a lot of ceremony / boilerplate associated with them (relative to the actual example code). For example:
https://github.com/flutter/flutter/blob/master/examples/api/lib/rendering/sliver_grid/sliver_grid_delegate_with_fixed_cross_axis_count.0.dart

We don't need so much ceremony around declaring a new `StatelessWidget`, passing a `_title`, etc., do we? I get that we want to show best practices, but I worry that in situ examples like this make it hard for a learner to see the wood from the trees. @hixie mentions that there's some things we should keep like putting the example in a widget and `runApp()`ing that widget, so that hot reload works, but that the current boilerplate is mostly a historical accident.

Can we trim our basic template a little?

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds some simple templates to the api examples to make it easier to write new tests for them.

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/96492

## Tests
 - Added example tests as well.

### CODE REVIEW

File: examples/api/README.md [hunk 1]
```diff
@@ -74,21 +74,28 @@ wanted to add a new example to the `Curve2D` class:
...
-to, in lower_snake_case, with an index relating to their order within the doc
-comment. So, for the `Curve2D` case, since it's in the `animation` package, in
-a file called `curves.dart`, and it's the first example, it goes in
+appropriate path under [`examples/api`](.). See the
+[sample_templates](./lib/sample_templates/) directory for examples of different
+types of samples with some best practices applied.
+
+The filename should match the location of the source file it is linked from,
+and is named for the symbol it is attached to, in lower_snake_case, with an
+index relating to their order within the doc comment. So, for the `Curve2D`
```
Comment (HansMuller) [inline examples/api/README.md:90]: So, for a `Curve2D` example, ...

File: examples/api/README.md [hunk 2]
```diff
@@ -123,18 +130,20 @@ context that the analyzer uses.
...
+builds and runs all the API examples, just to make sure that they start up
+without crashing. In addition, we also require writing tests of functionality in
+the examples, and generally just do what we normally do for writing tests. The
+one thing that makes it more challenging for the examples is that they can't
+really be written for testability in any obvious way, since that would probably
+complicate the example and make it harder to explain.
 
 As an example, in regular framework code, you might include a parameter for a
 `Platform` object that can be overridden by a test to supply a dummy platform,
 but in the example, this would be unnecessary complexity for the example. In all
```
Comment (HansMuller) [inline examples/api/README.md:143]: This would be unnecessarily complex for an example.

File: examples/api/README.md [hunk 3]
```diff
@@ -123,18 +130,20 @@ context that the analyzer uses.
...
-Examples are required to have tests. There is already a "smoke test" that runs
-all the API examples, just to make sure that they start up without crashing. In
-addition, we also require writing tests of functionality in the examples, and
-generally just do what we normally do for writing tests. The one thing that
-makes it more challenging for the examples is that they can't really be written
-for testability in any obvious way, since that would probably complicate the
-example and make it harder to explain.
+Examples are required to have tests. There is already a "smoke test" that simply
+builds and runs all the API examples, just to make sure that they start up
+without crashing. In addition, we also require writing tests of functionality in
```
Comment (HansMuller) [inline examples/api/README.md:135]:
```suggestion
without crashing. We also require functionality tests for
```

File: examples/api/README.md [hunk 4]
```diff
@@ -123,18 +130,20 @@ context that the analyzer uses.
...
+the examples, and generally just do what we normally do for writing tests. The
```
Comment (HansMuller) [inline examples/api/README.md:136]:
```suggestion
each example. Generally speaking, we just do what we normally do when writing tests. The
```

**Review by HansMuller — APPROVED** [review]
LGTM

File: examples/api/README.md (hunk 2 above)
Comment (gspencergoog) [reply examples/api/README.md:143]: Done.

File: examples/api/README.md (hunk 4 above)
Comment (gspencergoog) [reply examples/api/README.md:136]: Done

File: examples/api/README.md (hunk 3 above)
Comment (gspencergoog) [reply examples/api/README.md:135]: Done

**Pull request merged**